### ISSUE TITLE
Can't access admin page when using DuckDns + can't get the sensor working

### ISSUE THREAD

**Issue by Kuglikrug** [hassio-addons/addon-pi-hole#86]
# Problem/Motivation

Cannot access the pi-hole admin page while connected to hass.io via DuckDNS, works fine when trying to access over hassio.local:4865 when on the same network. Also i struggle setting up the pi-hole sensor, it's unavailable.

## Expected behavior

Access the Pi-Hole admin page and use the pi-hole sensor when i'm not home.

## Actual behavior

can only access the admin page when on the same network via hassio.local:4865

## Steps to reproduce

this is my add-on config:

> {
>   "password": "pw",
>   "update_lists_on_start": false,
>   "admin_port": 4865,
>   "dns_port": 53,
>   "ssl": true,
>   "certfile": "fullchain.pem",
>   "keyfile": "privkey.pem",
>   "interface": "",
>   "ipv6": true,
>   "ipv4_address": "",
>   "ipv6_address": "",
>   "virtual_host": "https://blablabla.duckdns.org",
>   "hosts": []
> }

this is my sensor config:

>- platform: pi_hole
>    host: localhost:4865
>    monitored_conditions:
>       - ads_blocked_today
>       - dns_queries_today
>       - unique_clients

**Comment by gcorgnet** [comment]
Hi, I'm encountering the same issue. It seems like I have been able to set up caddy with most of my other add-ons which required it so I can use [add-on-name].mydomain.duckdns.org.
Note: no port here, obviously since caddy takes care of that.

This doesn't work with pi-hole unfortunately and when trying to browse to
https://pihole.mydomain.duckdns.org, seems like the browser gets redirected to https://pihole.mydomain.duckdns.org:4865/admin

When I saw the virtual_host option, I was hopeful but it doesn't seem to do anything.
